File: Minimum Platforms Required.c
#include <stdio.h>
#include <stdlib.h>

int cmp(const void* a, const void* b) {
    return (*(int*)a - *(int*)b);
}

int findPlatform(int arr[], int dep[], int n) {
    qsort(arr, n, sizeof(int), cmp);
    qsort(dep, n, sizeof(int), cmp);

    int plat = 1, result = 1;
    int i = 1, j = 0;

    while (i < n && j < n) {
        if (arr[i] <= dep[j]) {
            plat++;
            i++;
        } else {
            plat--;
            j++;
        }
        if (plat > result) result = plat;
    }
    return result;
}

File: Minimum Platforms Required.py
def findPlatform(arr, dep):
    arr.sort()
    dep.sort()

    i = j = 0
    platforms = result = 0

    while i < len(arr) and j < len(dep):
        if arr[i] <= dep[j]:
            platforms += 1
            i += 1
        else:
            platforms -= 1
            j += 1
        result = max(result, platforms)

    return result
